internal/provider/provider.go

```diff
@@ -442,8 +478,46 @@ func providerConfigure(p *schema.Provider) schema.ConfigureContextFunc {
 	}
 }
 
+// buildClient is used to configure behavioral aspects of the provider. To configure the
+// cloud environment and authentication-related settings, use the providerConfigure function.
 func buildClient(ctx context.Context, p *schema.Provider, d *schema.ResourceData, authConfig *auth.Credentials) (*clients.Client, diag.Diagnostics) {
-	skipProviderRegistration := d.Get("skip_provider_registration").(bool)
+	// TODO: Remove this hardcoded default in v4.0
```

Comment:
Is this comment correct? should this be 
```suggestion
	// TODO: Remove this hardcoded default in v5.0
```

Reply:
This is here to provide a sane default when running in v3.x and the `resource_provider_registrations` property is not in the provider schema. In v4.0 (or beta mode) we can use the `Default` attribute value from the schema.